src/js.c

```diff
@@ -2075,12 +2087,20 @@ _hbTimerFired(natsTimer *timer, void* closure)
     if (oc)
     {
         natsSub_Lock(sub);
+        if (sub->libDlvWorker != NULL)
+        {
+            natsMutex_Lock(sub->libDlvWorker->lock);
+        }
         if (!sub->closed)
```

Comment:
Humm... there are may places where we check this under the sub's lock only. Sure, it is set under both locks and checked only under the lib's lock in `nats.c` (the code that deals with the delivery's pool thread), but if we find many other reports, then I think it would be simpler to modify `_deliverMsgs` so that we capture `sub->closed` under the sub's lock after releasing the delivery's lock. Anyway, for now change is ok, but again, let's keep that in mind.